### PR TITLE
validator: add FTS e2e test suite

### PR DESCRIPTION
**Pull request by knowack1**
Adds a comprehensive end-to-end test suite for full-text search (FTS) functionality in the validator. The new `fts` module covers index lifecycle management, BM25 search queries (basic, boolean AND/OR/NOT, phrase, and empty-result edge cases), and CRUD visibility — verifying that inserted, updated, and deleted documents are correctly reflected in search results.

To support FTS index creation, the shared `CreateIndexQuery` helper in `common.rs` is generalized to accept an arbitrary index type and make the `WITH OPTIONS` clause optional, which was previously hardcoded for vector indexes only.

Refs: VECTOR-636

### CODE REVIEW

File: crates/validator/src/fts.rs [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    let pks = fixture.extract_pks(&result);
+
+    assert_eq!(pks, vec![1], "updated term 'beta' should be searchable");
+
+    let result = get_opt_query_results(fixture.bm25_select_query("alpha"), &fixture.session).await;
+
+    assert!(
+        result.expect("failed to get query results").rows_num() == 0,
+        "old term 'alpha' should no longer match"
+    );
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/fts.rs]: Fixed

File: crates/validator/src/fts.rs [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+        .await
+    }
+
+    fn bm25_select_query_with_limit(&self, query: &str, limit: usize) -> String {
+        let table = &self.table;
+        format!(
+            "SELECT pk FROM {table} WHERE BM25(content, '{query}') > 0 \
+             ORDER BY BM25(content, '{query}') LIMIT {limit}"
+        )
+    }
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/fts.rs:100]: To be added when needed .

File: crates/validator/src/fts.rs [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+    fn bm25_select_query(&self, query: &str) -> String {
+        self.bm25_select_query_with_limit(query, 100)
+    }
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/fts.rs]: fixed

File: crates/validator/src/fts.rs [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+}
+
+impl e2etest::Fixture for Fixture {
+    async fn setup(setup: &mut impl e2etest::Setup) -> Self {
+        setup.setup::<TestActors>().await;
+        let actors = setup.get::<TestActors>().await.unwrap();
+        init(&actors).await;
+        let (session, _clients, keyspace, table, _index) = Self::init_fts_table(&actors).await;
+        Self {
+            actors,
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/fts.rs]: All test works this way.

File: crates/validator/src/fts.rs [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+        let keyspace = create_keyspace(&session).await;
+        let table = create_table(&session, "pk INT PRIMARY KEY, content TEXT", None).await;
+
+        let index = create_index(
+            CreateIndexQuery::new(&session, &clients, &table, "content")
+                .index_type("fulltext_index"),
+        )
+        .await;
+
+        (session, clients, keyspace, table, index)
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/fts.rs:136]: This is done in purpose. We are waiting for search to succeed.

File: crates/validator/src/common.rs [hunk 6]
```diff
@@ -679,17 +684,18 @@ pub struct CreateIndexQuery<'a> {
...
 
 impl<'a> CreateIndexQuery<'a> {
     pub fn new(
         session: &'a Session,
         clients: &'a [HttpClient],
         table: impl AsRef<str>,
-        vector_column: impl AsRef<str>,
+        target_column: impl AsRef<str>,
     ) -> Self {
         assert!(!clients.is_empty(), "No vector store clients provided");
```
[the remark this answers is not in the document]
Comment (knowack1) [reply crates/validator/src/common.rs:700]: vector store is the name of the service

**Review by ewienik — APPROVED** [review]
